src/main/java/dyorgio/runtime/macos/trayicon/fixer/jna/foundation/NSObject.java

```diff
@@ -39,28 +39,33 @@
     protected static final Pointer allocSel = Foundation.INSTANCE.sel_registerName("alloc");
     protected static final Pointer initSel = Foundation.INSTANCE.sel_registerName("init");
     protected static final Pointer releaseSel = Foundation.INSTANCE.sel_registerName("release");
+    protected static final Pointer retainSel = Foundation.INSTANCE.sel_registerName("retain");
     protected static final Pointer performSelectorOnMainThread$withObject$waitUntilDoneSel
             = Foundation.INSTANCE.sel_registerName("performSelectorOnMainThread:withObject:waitUntilDone:");
 
     final NativeLong id;
 
     public NSObject(NativeLong id) {
         this.id = id;
+        Foundation.INSTANCE.objc_msgSend(id, retainSel);
     }
 
     public final NativeLong getId() {
         return id;
     }
 
-    public void release() {
+    private void release() {
         Foundation.INSTANCE.objc_msgSend(id, releaseSel);
     }
 
     @Override
     @SuppressWarnings("FinalizeDeclaration")
     protected void finalize() throws Throwable {
-        release();
-        super.finalize();
+        try {
```

Comment:
I was thinking about retain/release... they need to exist, but not called automagically, only by developer.
Maybe some control can be done, ex: If retain is called, and release not before finalize, then call release.